packages/block-library/src/navigation-link/index.php

```diff
@@ -312,11 +312,45 @@ function build_variation_for_navigation_link( $entity, $kind ) {
 	$title       = '';
 	$description = '';
 
+	// Get default labels based on entity type
+	$default_labels = null;
+	if ( $entity instanceof WP_Post_Type ) {
+		$default_labels = WP_Post_Type::get_default_labels();
+	} elseif ( $entity instanceof WP_Taxonomy ) {
+		$default_labels = WP_Taxonomy::get_default_labels();
+	}
+
+	// Get title and check if it's default
+	$title            = '';
+	$is_default_title = false;
 	if ( property_exists( $entity->labels, 'item_link' ) ) {
 		$title = $entity->labels->item_link;
+		if ( $default_labels && isset( $default_labels['item_link'] ) ) {
+			$is_default_title = in_array( $title, $default_labels['item_link'], true );
+		}
 	}
+
+	// Get description and check if it's default
+	$description            = '';
+	$is_default_description = false;
 	if ( property_exists( $entity->labels, 'item_link_description' ) ) {
 		$description = $entity->labels->item_link_description;
+		if ( $default_labels && isset( $default_labels['item_link_description'] ) ) {
+			$is_default_description = in_array( $description, $default_labels['item_link_description'], true );
+		}
+	}
+
+	// Calculate singular name once (used for both title and description)
+	$singular = isset( $entity->labels->singular_name ) ? $entity->labels->singular_name : ucfirst( $entity->name );
+
+	// Set default title if needed
+	if ( $is_default_title || '' === $title ) {
+		$title = sprintf( /* translators: %s: Singular label of the entity. */ __( '%s Link' ), $singular );
```

Comment:
I thought the translators comment was meant to go on the previous line?

Reply:
```suggestion
		$title = sprintf( /* translators: %s: Singular label of the entity. */ __( '%s link' ), $singular );
```
I think we prefer sentence case.